### DISCUSSION TITLE
Feature collection link to GeoJSON items does not include `f=json` in href

### DISCUSSION THREAD

**Discussion by mdastous-bentley** [developmentseed/tipg#169] [General]
Hi,

I'm relatively new to OGC Features, and I'm currently trying to implement a OGC Feature client.  
I'm using Tipg as my test server and just noticed that collection links seem invalid.  For example here is the result for the sample countries dataset:
`http://localhost:8081/collections/public.countries?f=json`

```
{
  "id": "public.countries",
  "title": "public.countries",
  "links": [
    {
      "href": "http://localhost:8081/collections/public.countries",
      "rel": "self",
      "type": "application/json"
    },
    {
      "href": "http://localhost:8081/collections/public.countries/items",
      "rel": "items",
      "type": "application/geo+json",
      "title": "Items"
    },
    {
      "href": "http://localhost:8081/collections/public.countries/items?f=csv",
      "rel": "alternate",
      "type": "text/csv",
      "title": "Items (CSV)"
    },
    {
      "href": "http://localhost:8081/collections/public.countries/items?f=geojsonseq",
      "rel": "alternate",
      "type": "application/geo+json-seq",
      "title": "Items (GeoJSONSeq)"
    },
    {
      "href": "http://localhost:8081/collections/public.countries/queryables",
      "rel": "queryables",
      "type": "application/schema+json",
      "title": "Queryables"
    }
  ],
  "extent": {
    "spatial": {
      "bbox": [
        [
          -180.0,
          -89.99893188476562,
          180.0,
          83.599609375
        ]
      ],
      "crs": "http://www.opengis.net/def/crs/OGC/1.3/CRS84"
    }
  },
  "itemType": "feature",
  "crs": [
    "http://www.opengis.net/def/crs/OGC/1.3/CRS84"
  ]
}
```

It looks like the `items` link for `application/geo+json` type should be:
`http://localhost:8081/collections/public.countries/items?f=geojson`
instead of:
`http://localhost:8081/collections/public.countries/items`
which in fact returns an HTML format.

Or maybe I'm missing something?

Thanks

**Comment by vincentsarago** [comment]
@mdastous-bentley thank's for the report, I agree that we should have `f=geojson` there

**Reply by vincentsarago** [reply]
also I think by default without any `accept` headers the response should be of type `application/geo+json` not html 

`http://localhost:8081/collections/public.countries/items` will only return an HTML if used in a browser

**Reply by mdastous-bentley** [reply]
Do you mean the `self` link here? If so, it currently return `html` in the browser but advertised as `application/json`

**Reply by vincentsarago** [reply]
both `self` 

```
    {
      "href": "http://localhost:8081/collections/public.countries",
      "rel": "self",
      "type": "application/json"
    }
```

and `items`
```
   {
      "href": "http://localhost:8081/collections/public.countries/items",
      "rel": "items",
      "type": "application/geo+json",
      "title": "Items"
    }
```

have default to `json` and `geo+json` (you can verify this when using `curl`) 

```
# COLLECTION
curl http://127.0.0.1:8000/collections/public.countries --verbose                      
*   Trying 127.0.0.1:8000...
* Connected to 127.0.0.1 (127.0.0.1) port 8000 (#0)
> GET /collections/public.countries HTTP/1.1
> Host: 127.0.0.1:8000
> User-Agent: curl/8.1.2
> Accept: */*
> 
< HTTP/1.1 200 OK
< date: Tue, 05 Mar 2024 13:48:15 GMT
< server: uvicorn
< content-length: 961
< content-type: application/json
< cache-control: public, max-age=3600
< 
* Connection #0 to host 127.0.0.1 left intact

# ITEMS
curl http://127.0.0.1:8000/collections/public.countries/items --verbose
*   Trying 127.0.0.1:8000...
* Connected to 127.0.0.1 (127.0.0.1) port 8000 (#0)
> GET /collections/public.countries/items HTTP/1.1
> Host: 127.0.0.1:8000
> User-Agent: curl/8.1.2
> Accept: */*
> 
< HTTP/1.1 200 OK
< date: Tue, 05 Mar 2024 13:48:44 GMT
< server: uvicorn
< content-length: 105743
< content-type: application/geo+json
< cache-control: public, max-age=3600
< 
```

Both links are advertised with their defaults, but in browser, a header `Accept: text/html` will be automatically added, and thus `tipg` endpoint will return *html* content (instead of the default json type).

`tipg` output format depends on two things:
- query-parameter: `f=html` 
- headers: `Accept: text/html` 

I think this is not well described in the openapi documentation 

ref: https://docs.ogc.org/is/17-069r4/17-069r4.html#encodings

**Reply by mdastous-bentley** [reply]
oh I see, this is probably going to be an infinite source of confusion.
